pkg/controller/handlers/threads/threads.go

```diff
@@ -287,3 +289,23 @@ func (t *Handler) CleanupEphemeralThreads(req router.Request, _ router.Response)
 
 	return kclient.IgnoreNotFound(req.Delete(thread))
 }
+
+func (t *Handler) ActivateRuns(req router.Request, _ router.Response) error {
+	var runs v1.RunList
+	// This must be uncached since inactive things aren't in the cache.
+	if err := req.List(untriggered.UncachedList(&runs), &kclient.ListOptions{
+		Namespace:     req.Namespace,
+		FieldSelector: fields.SelectorFromSet(map[string]string{"spec.threadName": req.Object.GetName()}),
+	}); err != nil {
+		return fmt.Errorf("failed to list runs for thread %s: %w", req.Object.GetName(), err)
+	}
+
+	for _, run := range runs.Items {
+		v1.SetActive(&run)
```

Comment:
worth to check so that if it is already active, then don't update it?

Reply:
Yes, definitely.

Reply:
Fixed.